ldms/src/ldmsd/ldmsd.c

```diff
@@ -1804,11 +1830,111 @@ void ldmsd_str_list_destroy(struct ldmsd_str_list *list)
 	}
 }
 
+/* The -x arguments must be fully processed only after all args have
+ * first been parsed in ldmsd_process_cmd_line_arg to ensure correct
+ * behavior of -a and -A default settings.
+ */
+static int ldmsd_process_cmd_line_listener(const char *optarg)
+{
+	char *dup_xtuple = strdup(optarg);
+	char *dup_atuple = strdup(optarg);
+	if (!dup_xtuple || !dup_atuple) {
+		ldmsd_log(LDMSD_LERROR,"out of memory parsing -x arguments\n");
+		free(dup_xtuple);
+		free(dup_atuple);
+		return ENOMEM;
+	}
+	char *_xprt, *_port, *_host, *_auth = NULL, *x_auth_opts = NULL;
+	_auth = strchr(dup_atuple, '@');
+	if (_auth) {
+		dup_xtuple[_auth - dup_atuple] = '\0';
+		_auth++;
+	}
+	_xprt = dup_xtuple;
+	_port = strchr(dup_xtuple, ':');
+	if (!_port || _port[1] == ':') {
+		ldmsd_log(LDMSD_LERROR, "Bad -x port format, expecting at least "
+			"XPRT:PORT, but got: %s\n", optarg);
+		free(dup_xtuple);
+		free(dup_atuple);
+		return EINVAL;
+	}
+	*_port = '\0';
+	_port++;
+	/* optional `host` */
+	_host = strchr(_port, ':');
+	if (_host) {
+		*_host = '\0';
+		_host++;
+	}
+	/* optional `auth' follows @ (allow ipv6 host) */
+	if (_auth) {
+		/* optional `x_auth_opts` */
+		x_auth_opts = strchr(_auth, ':');
+		if (x_auth_opts) {
+			*x_auth_opts = '\0';
+			x_auth_opts++;
+		}
+	}
+	if (_host && !strlen(_host)) {
+		_host = NULL;
+	}
+	if (_auth && !strlen(_auth)) {
+		_auth = NULL;
+	}
+	/* Use the default auth domain unless specified */
+	if (!_auth) {
+		_auth = DEFAULT_AUTH_NAME;
```

Comment:
This should be
```
_auth = DEFAULT_AUTH
```
`DEFAULT_AUTH` is the name of the default authentication domain, which uses the authentication method `DEFAULT_AUTH_NAME`. Currently, `DEFAULT_AUTH_NAME` is `none`.

This results in users cannot start LDMSD with a listening endpoint using the default authentication method. LDMSD won't start with the following command, for example,
```
ldmsd -x sock:10001
```

Moreover, any user-specific default authentication method won't work.
```
ldmsd -x sock:10001 -a munge
```
This is the cause that the patch fails the GitHub test.